x/escrow/init.go

```diff
@@ -0,0 +1,62 @@
+package escrow
+
+import (
+	"encoding/hex"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/pkg/errors"
+)
+
+var _ weave.Initializer = (*Initializer)(nil)
+var burnAddress, _ = hex.DecodeString("0000000000000000000000000000000000000000")
+
+// Initializer fulfils the Initializer interface to load data from the genesis file
+type Initializer struct {
+	Minter cash.CoinMinter
+}
+
+// FromGenesis will parse initial escrow  info from genesis and save it in the database.
+func (i *Initializer) FromGenesis(opts weave.Options, db weave.KVStore) error {
+	var escrows []struct {
+		Sender    weave.Address   `json:"sender"`
+		Arbiter   weave.Condition `json:"arbiter"`
+		Recipient weave.Address   `json:"recipient"`
+		Timeout   int64           `json:"timeout"`
+		Amount    []*coin.Coin    `json:"amount"`
+	}
+
+	if err := opts.ReadOptions("escrow", &escrows); err != nil {
+		return err
+	}
+
+	bucket := NewBucket()
+	for j, e := range escrows {
+		escr := Escrow{
+			Sender:    e.Sender,
+			Arbiter:   e.Arbiter,
+			Recipient: e.Recipient,
+			Timeout:   e.Timeout,
+			Amount:    e.Amount,
+		}
+		if err := escr.Validate(); err != nil {
+			return errors.Wrapf(err, "invalid escrow at position: %d ", j)
+		}
+		if !weave.Address(escr.Sender).Equals(burnAddress) {
```

Comment:
This is a nice check for this use case, but...

What if I wanted to use a genesis escrow as, say, a programmatic vesting period?

I could for example, set a timeout in 12 months, the employees address as sender, burn address as recipient, and some trusted arbiter to burn it if they quit before 12 months, otherwise, they get it.

I think this check is very use-case dependent. We should document suggested uses but not force them

Reply:
Ok, got it.